From 556e1d93acd4fe7f983478e515e95abba97ee855 Mon Sep 17 00:00:00 2001
From: Kriti Pachhandara <kriti.pachhandara@intel.com>
Date: Mon, 15 Dec 2014 14:40:35 +0200
Subject: [PATCH 183/196] atomisp2: Set digital zoom true in yuvpp pipeline for
 SOC sensors

This patch enables the enable_dz to true for the SOC
sensors in video/yuvpp pipeline in moorefield.
This is done so that CSS API changes for YUV format
is not required.

Change-Id: Ia73462ef7ee6c5ec559ccb59db8d3b785112946c
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-11370
Signed-off-by: Kriti Pachhandara <kriti.pachhandara@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/32808
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c       | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 0d47f09..a3bb13a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -727,6 +727,8 @@ static void __apply_additional_pipe_config(
 		    .enable_reduced_pipe = true;
 		stream_env->pipe_configs[pipe_id]
 		    .enable_dz = false;
+		if (ATOMISP_SOC_CAMERA(asd))
+			stream_env->pipe_configs[pipe_id].enable_dz = true;
 
 		if (asd->params.video_dis_en) {
 			stream_env->pipe_extra_configs[pipe_id]
@@ -740,7 +742,10 @@ static void __apply_additional_pipe_config(
 		break;
 	case IA_CSS_PIPE_ID_YUVPP:
 	case IA_CSS_PIPE_ID_COPY:
-		stream_env->pipe_configs[pipe_id].enable_dz = false;
+		if (ATOMISP_SOC_CAMERA(asd))
+			stream_env->pipe_configs[pipe_id].enable_dz = true;
+		else
+			stream_env->pipe_configs[pipe_id].enable_dz = false;
 		break;
 	case IA_CSS_PIPE_ID_ACC:
 		stream_env->pipe_configs[pipe_id].mode = IA_CSS_PIPE_MODE_ACC;
-- 
1.9.1

